research/src/pages/popup/popup.proposal.mdx

```diff
@@ -0,0 +1,342 @@
+
+<p class="todo">
+  Probably need to specify some way of storing the relationship once subject is activated, for
+  purposes of deciding whether to hide a popup.
+</p>
+
+<p class="question">
+  Should we have a way to suppress these steps upon focus, so that the author can enable the popup
+  to be shown upon text entry (or some other logic) instead?
+</p>
+
+<h3 id="showing-popup-steps">
+  Showing a <code>popup</code> element steps
+</h3>
+
+1. Run the [hiding currently-shown `popup` elements steps](#hiding-current-popups).
+2. Add the _subject_ to _subject_'s
+   [node document](https://dom.spec.whatwg.org/#concept-node-document)'s
+   [top layer](https://fullscreen.spec.whatwg.org/#top-layer), at the top of the stack.
+3. Set the `open` IDL attribute on the _subject_ to `true`.
+
+<h4 id="hiding-current-popups">
+  Hiding currently-shown <code>popup</code> elements steps
+</h4>
+
+Let _candidate subject_ be the new candidate `popup` element to be shown.
+If applicable, let _activating element_ be the element whose `popup` attribute specifies the ID of
+the _candidate subject_, and which has been activated by the user.
+If applicable, let the _anchor element_ be the element whose ID is referenced by the `anchor`
+specified on the _candidate subject_.
+
+Starting at the bottom of the top-layer stack: for each `popup` element which has an `open` IDL
+attribute set to `true`, and which is not the _candidate subject_:
+
+1. Let _subject_ be the current `popup` element in the top-layer stack.
+2. Walk the DOM to determine if the _subject_ is an ancestor to the _candidate subject_, the
+   _activating element_, or the _anchor element_.
+3. If the _subject_ is an ancestor, then return.
+4. Otherwise, for this _subject_ and any other `popup`s higher in the stack, run the
+   [hiding a `popup` element steps](#hiding-current-popups).
+
+<p class="todo">
+  Step #2 probably needs to be more precise, and more specific about Shadow/light DOM
+</p>
+
+## Setting focus when the `popup` element is shown
+
+### The `delegatesfocus` attribute
+
+The `delegatesfocus` attribute is a
+[boolean attribute](https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#boolean-attribute).
+When specified, it indicates that when the `popup` element is shown, focus should move to the first
+focusable descendent of the `popup` element.
+
+**Example:**
+
+```html
+<popup delegatesfocus>
+  <p>I am not a focusable element.</p>
+  <p>Nor am I.</p>
+  <button>I will be focused whenever the popup becomes focused.</button>
+</popup>
+```
+
+### Focusing steps
+
+The popup [focusing steps](https://html.spec.whatwg.org/multipage/interaction.html#focusing-steps)
+for a `popup` element _subject_ are as follows:
+
+1. If the `delegatesfocus` attribute is specified on _subject_, let _control_ be the first focusable
+   descendant element of _subject_, in [tree order](https://dom.spec.whatwg.org/#concept-tree-order),
```

Comment:
So the spec for ["Get the focusable area"](https://html.spec.whatwg.org/multipage/interaction.html#get-the-focusable-area) should be used here, I think. And actually, I think it'd be a lot better if we folded this algorithm (focusing steps for `<popup>` and even `<dialog>`) back into the "Get the focusable area" spec. It'd be a lot cleaner, I think.

Reply:
Done in https://github.com/openui/open-ui/pull/355/commits/ecb4964478f3c7330f58914a79955000d906108c! I've created a new section of this document at the bottom for text that should land in the focusable area algorithm.